File: 0001_0100/0064/0064.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    s = "2"
    p = [True] * (20222)
    i = 3
    while i <= 20222:
        if p[i]:
            for j in range(i * i, 20222, 2 * i):
                p[j] = False
        i += 2
    for i in range(3, 20222, 2):
        if p[i]:
            s += str(i)
    q.write("".join(s[i - 1] for i in (int(x) for x in f.read().split()[1:])))
    
    

File: 0001_0100/0064/0064.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n, m int
	var p [20222]bool
	a := []string{"2"}
	for i := 3; i < 20222; i += 2 {
		if !p[i] {
			for j := i * i; j < 20222; j += 2 * i {
				p[j] = true
			}
			a = append(a, fmt.Sprint(i))
		}
	}
	s := strings.Join(a, "")
	fmt.Fscan(f, &n)
	for ; n > 0; n-- {
		fmt.Fscan(f, &m)
		fmt.Fprint(q, s[m-1:m])
	}
}
